xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -226,12 +231,39 @@ private boolean deleteToRecycleBin(XWikiContext context) throws XWikiException
             doesAffectChildren(request, doc.getDocumentReference()) ? doc.getDocumentReference().getLastSpaceReference()
                 : doc.getTranslatedDocument(context).getDocumentReferenceWithLocale();
 
-        return deleteToRecycleBin(documentReference, context);
+        return deleteToRecycleBin(documentReference, context, skipRecycleBin);
+    }
+
+    /**
+     *
+     * @param entityReference the entity to delete
+     * @param context the current content
+     * @return {@code true} if the user is redirected, {@code false} otherwise
+     * @throws XWikiException if anything goes wrong during the document deletion
+     * @deprecated since 12.8RC1, use {@link #deleteToRecycleBin(EntityReference, XWikiContext, boolean)} instead
+     */
+    @Deprecated
+    protected boolean deleteToRecycleBin(EntityReference entityReference, XWikiContext context)
+        throws XWikiException
+    {
+        return this.deleteToRecycleBin(entityReference, context, false);
```

Comment:
Do we use this method? If not it should be moved to legacy.

Reply:
I have kept it to avoid potential breaking changes. But AFAIK, it could be moved to legacy though I never did that.

Reply:
You need to search to verify it's not used and if is you need to change the call to the new API and then move to legacy.

Reply:
Inside of `xwiki-platform` we are ok. I was more afraid of classes inheriting `DeleteAction` on external modules.

Reply:
But but but, that's the point, we never break backward compat! :)

See https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HBackwardCompatibility

Reply:
Method moved to legacy https://github.com/xwiki/xwiki-platform/pull/1401/files#diff-a142e67ed19481336d36f2c6f66b2354